src/rbindlist.c

```diff
@@ -244,9 +244,8 @@ SEXP rbindlist(SEXP l, SEXP usenamesArg, SEXP fillArg, SEXP idcolArg, SEXP ignor
     ncol = length(VECTOR_ELT(l, first));  // ncol was increased as if fill=true, so reduce it back given fill=false (fill==false checked above)
   }
 
-  int nprotect = 0;
```

Comment:
I don't get the rationale behind this. IMO there should be no different between `UNPROTECT(nprotect); UNPROTECT(2)` in the end or `nprotect += 2; UNPROTECT(nprotect);`

Reply:
I like the clarity of a literal `2` -- I am worried that by always using the simple `UNPROTECT(1); nprotect++;` "trick" it makes it easier to obscure poor design where 100s of things wind up protected (#6258). My thinking is it's easier to catch this when we are forced to stare at the code a little more before deciding what to write in `UNPROTECT()`. I moved back to `2+nprotect`.

As to why `UNPROTECT(2); UNPROTECT(nprotect);` instead of `UNPROTECT(2+nprotect);`, well, that was just me trying to explore why rchk is giving us the current "error", turns out it's _not_ anything to do with mixing literal+variable there.

Reply:
Ok but then why not initialize with `nprotect = 2` and `UNPROTECT(nprotect)` in the end?

Reply:
Oh, great idea.